windows.mk

```diff
@@ -1,3 +1,238 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+#=======================================================================================================================
+# Default Paths
+#=======================================================================================================================
+
+!ifndef PREFIX
+PREFIX = $(USERPROFILE)
+!endif
+
+!ifndef INSTALL_PREFIX
+INSTALL_PREFIX = $(USERPROFILE)
+!endif
+
+!ifndef LD_LIBRARY_PATH
+LD_LIBRARY_PATH = $(USERPROFILE)/lib
+!endif
+
+#=======================================================================================================================
+# Build Configuration
+#=======================================================================================================================
+
+BUILD = release
+!if "$(DEBUG)" == "yes"
+BUILD = dev
+!endif
+
+#=======================================================================================================================
+# Project Directories
+#=======================================================================================================================
+
+!ifndef BINDIR
+BINDIR = $(MAKEDIR)\bin
+!endif
+
+!ifndef INCDIR
+INCDIR = $(MAKEDIR)\include
+!endif
+
+SRCDIR = $(MAKEDIR)\src
+
+BUILD_DIR = $(MAKEDIR)\target\release
+!if "$(BUILD)" == "dev"
+BUILD_DIR = $(MAKEDIR)\target\debug
+!endif
+
+#=======================================================================================================================
+# Toolchain Configuration
+#=======================================================================================================================
+
+# Rust
+!ifndef CARGO
+CARGO = $(USERPROFILE)\.cargo\bin\cargo.exe
+!endif
+CARGO_FLAGS = $(CARGO_FLAGS) --profile $(BUILD)
+
+#=======================================================================================================================
+# Libraries
+#=======================================================================================================================
+
+DEMIKERNEL_LIB = $(BUILD_DIR)/demikernel.dll.lib
+LIBS = $(DEMIKERNEL_LIB)
+
+#=======================================================================================================================
+# Build Parameters
+#=======================================================================================================================
+
+!ifndef LIBOS
+LIBOS = catnapw
```

Comment:
Are we having different LibOs names for different platforms?  I would think this would just be "catnap".

Reply:
I tried to match the source directory name to the LibOs name in this case as the directory structure for Catnap for Linux (_src/rust/catnap_) and Windows (_src/rust/catnapw_) are different. We can try to keep the name same by conditionally compiling different source trees, however I thought this might be a simpler approach.

Reply:
I don't have a strong opinion on this one.  We could also have a completely separate catfoo name, I suppose.  This probably all needs to be revisited later when we change things to allow multiple Demikernel I/O stacks to be active at the same time.  Right now, LibOses and I/O stacks are 1:1, which we presumably won't want to be the case in the future.